lib/model/country.dart

```diff
@@ -1,35 +1,67 @@
+import 'package:built_collection/built_collection.dart';
+import 'package:built_value/built_value.dart';
+import 'package:built_value/serializer.dart';
 import 'package:flutter/material.dart';
 import 'package:plante/l10n/strings.dart';
 import 'package:plante/logging/log.dart';
 
-@immutable
-class Country {
-  static const ANDORRA = 'ad';
-  static const UNITED_ARAB_EMIRATES = 'ae';
-  static const AFGHANISTAN = 'af';
-  static const ANTIGUA_AND_BARBUDA = 'ag';
-  static const BELGIUM = 'be';
-  static const NETHERLANDS = 'nl';
-  static const FRANCE = 'fr';
-  static const GERMANY = 'de';
-  static const RUSSIA = 'ru';
+part 'country.g.dart';
+
+class Country extends EnumClass {
+  ///Andorra
+  static const Country ad = _$ad;
+  ///United Arab Emirates
+  static const Country ae = _$ae;
+  ///Afghanistan
+  static const Country af = _$af;
+  ///Antigua and Barbuda
+  static const Country ag = _$ag;
+  ///Belgium
+  static const Country be = _$be;
+  ///Netherlands
+  static const Country nl = _$nl;
+  ///France
+  static const Country fr = _$fr;
+  ///Germany
+  static const Country de = _$de;
+  ///Luxembourg
+  static const Country lu = _$lu;
+  ///Russia
+  static const Country ru = _$ru;
+
+  const Country._(String name) : super(name);
+
+  static BuiltSet<Country> get values => _$values;
+  static Country valueOf(String name) => _$valueOf(name);
+  static Serializer<Country> get serializer => _$countrySerializer;
 
 // List of countries we load the products from OFF linked to a store
-  static const enabledCountryCodes = [BELGIUM, NETHERLANDS, GERMANY, FRANCE];
+  static const enabledCountryCodes = [be, nl, de, fr, lu];
```

Comment:
yes, i moved it from OffShopsManager to Country. Mainly in the back of my head to keep country logic together, also if there might be a future purpose where we also only want to roll out to a specific countrylist.

Reply:
I have next concern - if we have many features which have their specific set of enabled/disabled countries, the `Country` enum will get a lot more of such List-fields.
If you consider fields of a class as something that this class owns, then it starts to look a bit strange that `Country` owns a list of places where certain features should enabled/disabled.

To me, intuitively, it feels like if a feature wants to specify a list of countries where this feature is enabled, then the feature should create such a list in its package, rather than in a general-purpose class which is intended to use by all other code.

Imagine if there was a class `Char` with static fields `Char.charsForRussian` and `Char.charsForDutch` (like we have `Country.countriesForSuggestedOFFProducts`).
Instead of such a structuring, there should be classes like `RussianAlphabet` and `DutchAlphabet` with static fields `RussianAlphabet.chars` and `DutchAlphabet.chars`.

What do you think?

Reply:
No, i follow you. It is not the responsibility of the class Country to maintain that list. I have already moved it :-)